python/tchannel/handler.py

```diff
@@ -80,9 +80,14 @@ class TChannelRequest(object):
 
     def __init__(self, context, conn):
         self.message = context.message
-        self.header = getattr(self.message, 'arg_2', None)
-        self.body = getattr(self.message, 'arg_3', None)
-        self.method = getattr(self.message, 'arg_1', None)
+
+        try:
+            self.method = self.message.args[0]
+            self.header = self.message.args[1]
+            self.body = self.message.args[2]
+        except:
+            pass
```

Comment:
This is extremely fragile. Check the length of `args` and handle the cases explicitly.